R/match_on_ltcs.R

```diff
@@ -0,0 +1,100 @@
+#' Match on LTC changed_dob and dates of LTC incidence (based on hospital incidence only)
+#'
+#' @description Match on LTC changed_dob and dates of LTC incidence (based on hospital incidence only)
+#'
+#' @param
+#'
+#' @return
+#' @export
+#'
+#' @examples
+match_on_ltcs <- function(data, year) {
+  # Match on LTC lookup
+  matched <- dplyr::left_join(
+    data,
+    readr::read_rds(get_ltcs_path(year)),
+    by = "chi",
+    suffix = c("", "_ltc")
+  ) %>%
+    # Replace any NA values with 0 for the LTC flags
+    dplyr::mutate(dplyr::across("arth":"digestive", ~ tidyr::replace_na(., 0)))
+  matched <- correct_demographics(matched, year)
+  return(matched)
+}
+
+correct_demographics <- function(data, year) {
+  # Checking and changing DOB and age
+  # data <- data %>%
+  test <- data %>%
+    dplyr::filter(!is_missing(chi)) %>%
+    dplyr::mutate(
+      # Create a dob in the previous century from the chi number
+      chi_dob_min = phsmethods::dob_from_chi(
+        chi_number = chi,
+        chi_check = FALSE,
+        min_date = lubridate::ymd("1900-01-01"),
+        max_date = lubridate::ymd("1999-12-31")
+      ),
+      # Create a dob in the current century from chi (will return NA if in the future)
+      chi_dob_max = phsmethods::dob_from_chi(
+        chi_number = chi,
+        chi_check = FALSE,
+        min_date = lubridate::ymd("2000-01-01"),
+        max_date = end_fy(year)
```

Comment:
`max_date` will always be the start of the activity as we know they have to have been born on or before the activity started.

```suggestion
        max_date = pmin(record_keydate1, lubridate::ymd("1999-12-31"))
      ),
      # Create a dob in the current century from chi (will return NA if in the future)
      chi_dob_max = phsmethods::dob_from_chi(
        chi_number = chi,
        chi_check = FALSE,
        min_date = lubridate::ymd("2000-01-01"),
        max_date = record_keydate1
```

Reply:
`phsmethods::dob_from_chi` requires `min_date <= max_date` and do not allow `NA`. The column `keydate1_dateformat` contains `NA` in some rows.

So, for line 36, I would write 
```
      max_date = pmin(
        dplyr::if_else(
          is.na(keydate1_dateformat),
          lubridate::ymd("1999-12-31"),
          keydate1_dateformat
        ),
        lubridate::ymd("1999-12-31")
      )
```

For line 43, `keydate1_dateformat`  ( which is essentially record_keydate1 but with an appropriate date format, is it?) can be earlier than `2000-01-01` which is specified as `min_date`. So `dob_from_chi` cannot allow this. Therefore, I guess it would better keep it as ` max_date = keydate1_dateformat`.

Happy to discuss. Thanks